### PR TITLE
Merge Registry into Driver

### PR DESCRIPTION
**Pull request by kelunik**
See #85.

### CODE REVIEW

File: src/Loop/Driver.php [hunk 1]
```diff
@@ -13,7 +18,7 @@
      *
      * @return void
      */
-    public function run();
+    public abstract function run();
```
Comment (joelwurtz) [inline src/Loop/Driver.php]: it should be `abstract public` for PSR2 compliance (same for others of course :) )

Comment (kelunik) [inline src/Loop/Driver.php]: I hate PSR-2.

Comment (geggleto) [inline src/Loop/Driver.php]: agreed.

Comment (kelunik) [inline src/Loop/Driver.php]: Fixed in the latest commit.

File: src/Loop/Driver.php [hunk 2]
```diff
@@ -189,19 +194,29 @@ public function unreference($watcherId);
      *
      * @return void
      */
-    public function storeState($key, $value);
+    public final function storeState($key, $value)
```
Comment (joelwurtz) [inline src/Loop/Driver.php]: I think you will hate me, but this is also not PSR2 compliant (should be `final public`), i'm sorry, but if one day if see you in person i will offer you a beer to compensate :)

(Same for fetchState)

Comment (kelunik) [inline src/Loop/Driver.php]: Why should I hate you? It's only PSR-2. Fixed, thanks. :-)

Actually, I remembered I should do this, but forgot to do so after I opened the editor.

**Pull request merged**